src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -22,58 +21,172 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
-                if (featureFlag.Conditions?.ClientFilters == null || !featureFlag.Conditions.ClientFilters.Any()) // workaround since we are not yet setting client filters to null
-                {
-                    //
-                    // Always on
-                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", true.ToString()));
-                }
-                else
+                using (JsonDocument document = JsonDocument.Parse(setting.Value))
                 {
-                    //
-                    // Conditionally on based on feature filters
-                    for (int i = 0; i < featureFlag.Conditions.ClientFilters.Count; i++)
-                    {
-                        ClientFilter clientFilter = featureFlag.Conditions.ClientFilters[i];
+                    JsonElement root = document.RootElement;
 
-                        _featureFilterTracing.UpdateFeatureFilterTracing(clientFilter.Name);
+                    bool isFeatureEnabled = false;
 
-                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.EnabledFor}:{i}:Name", clientFilter.Name));
+                    if (root.TryGetProperty(FeatureManagementConstants.EnabledJsonPropertyName, out JsonElement enabledElement))
+                    {
+                        if (enabledElement.ValueKind == JsonValueKind.True || enabledElement.ValueKind == JsonValueKind.False)
+                        {
+                            isFeatureEnabled = enabledElement.GetBoolean();
+                        }
+                        else if (enabledElement.ValueKind == JsonValueKind.String && bool.TryParse(enabledElement.GetString(), out bool enabled))
+                        {
+                            isFeatureEnabled = enabled;
+                        }
+                        else
+                        {
+                            throw CreateFeatureFlagFormatException(
+                                FeatureManagementConstants.EnabledJsonPropertyName,
+                                setting.Key,
+                                enabledElement.ValueKind.ToString(),
+                                $"{JsonValueKind.True}' or '{JsonValueKind.False}");
+                        }
+                    }
+
+                    string id = null;
 
-                        foreach (KeyValuePair<string, string> kvp in new JsonFlattener().FlattenJson(clientFilter.Parameters))
+                    if (root.TryGetProperty(FeatureManagementConstants.IdJsonPropertyName, out JsonElement idElement))
+                    {
+                        if (idElement.ValueKind == JsonValueKind.String)
+                        {
+                            id = idElement.GetString();
+                        }
+                        else if (idElement.ValueKind != JsonValueKind.Null)
```

Comment:
Is this saying that it's okay for id to be null?

Reply:
I was trying to model exactly what JsonSerializer was doing previously, and from what I tested it would allow you to pass null for a string property even if that was unlikely to actually happen in practice.

Reply:
Got it. So this is for backward compatibility.

I think we should check id first and skip the feature flag if id is null.